api/api.go

```diff
@@ -115,67 +111,13 @@ func (a *API) QueryRange(w http.ResponseWriter, r *http.Request, _ httprouter.Pa
 			level.Error(a.logger).Log("err", err, "series", ls.String())
 		}
 
-		res.Series = append(res.Series, resSeries)
+		res = append(res, resSeries)
 	}
-	// TODO(bwplotka): Handle warnings.
 	if set.Err() != nil {
```

Comment:
the way to avoid typos here is to do

```suggestion
	if err := set.Err(); err != nil {
```
And use `err` in return (: